File: Neetcode/GroupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        # ana = {}
        # for s in strs:
        #     cntr = Counter(s)
        #     if cntr in ana.keys():
        #         ana[cntr].append(s)
        #     else:
        #         ana[cntr] = list()
        # ans = []
        # for k,v in ana.items():
        #     ans.append(v)
        # return ans
        
        dct = {}
        for s in strs:
            srt = ''.join(sorted(s))
            if srt in dct:
                dct[srt].append(s)
            else:
                dct[srt] = [s]
        ans = []
        for k,v in dct.items():
            ans.append(v)
        return ans

File: Neetcode/GroupAnagrams.cpp
#include <map>
#include <string>


class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> vect;
        map<map<char, int>, vector<string>> ans;
        for(string s: strs){
            map<char, int> count;
            for(char c: s){
                if (count.find(c) != count.end()){
                    count[c] += 1;
                }
                else{
                    count[c] = 1;
                }
            }
            ans[count].push_back(s);
        }
        for (auto itr=ans.begin(); itr!=ans.end(); itr++){
            vect.push_back(itr->second);
        }
        return vect;
    }
};
